include/emp/prefab/ControlPanel.hpp

```diff
@@ -0,0 +1,331 @@
+     * @param in_info info object associated with this component
+     */
+    ControlPanel(
+      const int & refresh_rate,
+      const std::string & refresh_unit,
+      web::internal::DivInfo * in_info
+    ) : web::Div(in_info),
+    toggle_run{
+      FontAwesomeIcon{"fa-play"}, FontAwesomeIcon{"fa-pause"},
+      "success", "warning",
+      true, false,
+      emp::to_string(GetID(), "_main_toggle")
+    },
+    button_line(ButtonGroup{emp::to_string(GetID(), "_main")}),
+    step{
+      [](){ ; },
+      "<span class=\"fa fa-step-forward\" aria-hidden=\"true\"></span>",
+      emp::to_string(GetID(), "_main_step")
+    }
+    {
+      AddAttr(
+        "class", "btn-toolbar",
+        "class", "space_groups",
+        "role", "toolbar",
+        "aria-label", "Toolbar with simulation controls"
+      );
+      SetRefreshRate(refresh_rate, refresh_unit);
+      step.AddAttr(
+        "class", "btn",
+        "class", "btn-success"
+      );
+
+      static_cast<Div>(*this) << button_line;
+      button_line << toggle_run;
+      button_line << step;
+
+      AddAnimation(GetID(),
+        [&run_sim=GetSimulation(),
+          &refresh_list=Info()->GetRefreshList(),
+          &do_redraw=Info()->GetRedrawChecker()]
+        (const web::Animate & anim) mutable {
+          // Run the simulation function every frame
+          run_sim();
+          // Redraw widgets according to a rule
+          if(do_redraw(anim)) {
+            for (auto & wid : refresh_list) {
+              wid.Redraw();
+            }
+          }
+        }
+      );
+
+      toggle_run.SetCallback(
+        [&anim=Animate(GetID()), step=web::Button(step)]
+        (bool is_active) mutable {
+          if (is_active) {
+            anim.Start();
+          } else {
+            anim.Stop();
+          }
+        }
+      );
+
+      step.SetCallback([&anim=Animate(GetID())]() {
+        anim.Step();
+      });
+    }
+
+    public:
+    /**
+     * Contructor for a Control panel.
+     * @param refresh_rate the number of milliseconds or frames between refreshes
+     * @param refresh_mode units of "MILLISECONDS" or "FRAMES"
+     * @param in_id HTML ID of control panel div
+     */
+    ControlPanel(
+      const int & refresh_rate,
+      const std::string & refresh_unit,
+      const std::string & in_id="")
+    : ControlPanel(
+      refresh_rate,
+      refresh_unit,
+      new internal::ControlPanelInfo(in_id)
+    ) { ; }
+
+    /**
+     * Set the simulation for this control panel
+     * @param sim the function to be run every frame (as fast as possible)
+     */
+    ControlPanel & SetSimulation(const std::function<void()> & sim) {
+      Info()->SetSimulation(sim);
+      return *this;
+    }
+
+    /**
+     * Get the simulation for this control panel
+     * @return the function to be run every frame (as fast as possible)
+     */
+    const std::function<void()> & GetSimulation() const {
+      return Info()->GetSimulation();
+    }
+
+    /**
```

Comment:
I would have this take a std::function and note that they can write their own or use ControlPanel::MillisecondRefresher or ControlPanel::FrameRefresher